test_e2e.sh

```diff
@@ -310,6 +310,42 @@ assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
 # Wrap up
 pass
 
+##############################################
+# Test worktree-cleanup
+##############################################
+testcase "worktree-cleanup"
+
+echo "$TESTCASE" > "$REPO"/file
+git -C "$REPO" commit -qam "$TESTCASE"
+GIT_SYNC \
+    --wait=20 \
+    --repo="file://$REPO" \
+    --branch=e2e-branch \
+    --rev=HEAD \
+    --root="$ROOT" \
+    --dest="link" \
+    > "$DIR"/log."$TESTCASE" 2>&1 &
+sync_pid=$!
+
+# wait for first sync
+sleep 5
+assert_link_exists "$ROOT"/link
+assert_file_exists "$ROOT"/link/file
+assert_file_eq "$ROOT"/link/file "$TESTCASE"
+
+# second commit
+echo "$TESTCASE-ok" > "$REPO"/file2
+git -C "$REPO" add file2
+git -C "$REPO" commit -qam "$TESTCASE new file"
+REV=$(git -C "$REPO" rev-list -n1 HEAD)
+git -C "$REPO" worktree add "$ROOT"/"$REV" --no-checkout
+sleep 25
```

Comment:
If we reduce --wait time, we don't need to sleep so long.

Reply:
Updated to wait 10s instead of 20s.